tfhe/src/integer/server_key/radix_parallel/add.rs

```diff
@@ -59,14 +59,37 @@ fn should_hillis_steele_propagation_be_faster(num_blocks: usize, num_threads: us
     };
 
     // Estimate the latency of the parallelized algorithm
-    let mut parallel_expected_latency = 2 * compute_latency_of_one_layer(num_blocks, num_threads);
-    let max_depth = num_blocks.ceil_ilog2();
-    let mut space = 1;
-    for _ in 0..max_depth {
-        let num_block_at_iter = num_blocks - space;
-        let iter_latency = compute_latency_of_one_layer(num_block_at_iter, num_threads);
-        parallel_expected_latency += iter_latency;
-        space *= 2;
+    // One pre-processing layer, one layer to compute what happens in each grouping,
+    // one final post processing layer
+    let mut parallel_expected_latency = 3 * compute_latency_of_one_layer(num_blocks, num_threads);
+
+    let grouping_size = full_modulus.ilog2();
+    let num_groups = num_blocks.div_ceil(grouping_size as usize);
+
+    let num_carry_to_resolve = num_groups.saturating_sub(1);
+
+    let sequential_depth = (num_carry_to_resolve.saturating_sub(1) as u32) / (grouping_size - 1);
```

Comment:
Yes, we remove one, because we don't need to resolve the carry of the last block

and then for the sequential depts we also remove one because the carry of the first block is already known so we don't count it whereas for the hillise-steele estimation, the firstg carry is part of the number of cells in the array